packages/compiler/src/server/completion.ts

```diff
@@ -411,31 +413,56 @@ function addIdentifierCompletion(
   for (const [key, { sym, label, suffix }] of result) {
     let kind: CompletionItemKind;
     let deprecated = false;
-    const node = getSymNode(sym);
-    const type = sym.type ?? program.checker.getTypeForNode(node);
+    const symNode = getSymNode(sym);
+    const type = sym.type ?? program.checker.getTypeForNode(symNode);
     if (sym.flags & (SymbolFlags.Function | SymbolFlags.Decorator)) {
       kind = CompletionItemKind.Function;
-    } else if (sym.flags & SymbolFlags.Namespace && node.kind !== SyntaxKind.NamespaceStatement) {
+    } else if (
+      sym.flags & SymbolFlags.Namespace &&
+      symNode.kind !== SyntaxKind.NamespaceStatement
+    ) {
       kind = CompletionItemKind.Module;
-    } else if (node?.kind === SyntaxKind.AliasStatement) {
+    } else if (symNode?.kind === SyntaxKind.AliasStatement) {
       kind = CompletionItemKind.Variable;
-      deprecated = getDeprecationDetails(program, node) !== undefined;
+      deprecated = getDeprecationDetails(program, symNode) !== undefined;
     } else {
       kind = getCompletionItemKind(program, type);
       deprecated = getDeprecationDetails(program, type) !== undefined;
     }
     const documentation = getSymbolDetails(program, sym);
-    const item: CompletionItem = {
-      label: label ?? key,
-      documentation: documentation
-        ? {
-            kind: MarkupKind.Markdown,
-            value: documentation,
-          }
-        : undefined,
-      kind,
-      insertText: printIdentifier(key) + (suffix ?? ""),
-    };
+    let edit: TextEdit | undefined;
+    const startWith$ = sym.name.startsWith("$");
+    if (startWith$) {
+      const targetNode = getSourceLocation(node);
```

Comment:
can we just check the source, if the source starts with $, just trimStart($) for insert?